tests/acceptance/helpers/filesFoldersHelpers.js

```diff
@@ -0,0 +1,50 @@
+const { client } = require('nightwatch-api')
+const { propfind } = require('./webdavHelper')
+const convert = require('xml-js')
+const { relativeTo, normalize, join } = require('./path')
+
+exports.getAllFilesFolders = function () {
+  const user = client.globals.currentUser
+  const backendURL = client.globals.backend_url
+  const baseURL = new URL(join(backendURL, `remote.php/dav/files/${user}/`))
+  const basePath = baseURL.pathname
+  return propfind(`/files/${user}`, user, [])
+    .then(str => {
+      const data = convert.xml2js(str, { compact: true })['d:multistatus']['d:response']
+      const filepaths = data.map(elements =>
+        normalize(
+          elements['d:href']._text)
+      )
+      const filenames = filepaths.map(element => relativeTo(basePath, element))
+      // The first element in the filenames is the username which is not actually a filename.
+      // The first element should therefore be removed from filenames
+      return filenames.splice(1)
+    })
+}
+
+exports.getAllFolders = async function () {
+  const filenames = await exports.getAllFilesFolders()
+  const folders = filenames.filter(elements => elements.endsWith('/'))
+  return folders.map(folder => folder.slice(0, -1))
+}
+
+exports.getAllFiles = async function () {
+  const filenames = await exports.getAllFilesFolders()
+  return filenames.filter(elements => !elements.endsWith('/'))
+}
+
+exports.getFilesMatchingPattern = async function (pattern) {
+  const files = await exports.getAllFiles()
+  return files.filter(elements => elements.toLowerCase().includes(pattern))
+}
+
+exports.getFoldersMatchingPattern = async function (pattern) {
+  const files = await exports.getAllFolders()
+  return files.filter(elements => elements.toLowerCase().includes(pattern))
+}
+
+exports.getFilesFoldersMatchingPattern = async function (pattern) {
+  const filesMatchingPattern = await exports.getFilesMatchingPattern(pattern)
+  const foldersMatchingPattern = await exports.getFoldersMatchingPattern(pattern)
+  return foldersMatchingPattern.concat(filesMatchingPattern)
```

Comment:
```
const files = getAllFilesFolders()
return files.filter(elements => elements.toLowerCase().includes(pattern)

Reply:
This cannot be used

Reply:
@jasson99, please explain why that's the case for the record.

Reply:
getAllFilesFolders returns folders with slash attached to them which has been formatted in getAllfolders